td-shim/src/bin/td-shim/shim_info.rs

```diff
@@ -35,7 +37,8 @@ impl BootTimeStatic {
     // Validate the metadata and get the basic infomation from
     // it if any
     pub fn new() -> Option<Self> {
-        let metadata_offset = unsafe { *((u32::MAX - TDX_METADATA_OFFSET + 1) as *const u32) };
+        let metadata_offset = unsafe { *((u32::MAX - TDX_METADATA_OFFSET + 1) as *const u32) }
+            - TD_SHIM_LARGE_PAYLOAD_SIZE;
```

Comment:
will this overflow?

Reply:
Added the `checked_sub` to prevent potential overflow, thanks